src/richie/apps/courses/templates/courses/cms/course_detail.html

```diff
@@ -11,9 +11,16 @@
 {% endblock meta_index_rules %}
 
 {% block meta_rdfa_context %}
-    <meta property="og:type" content="website" />
-    {% get_placeholder_plugins "course_cover" as plugins %}
-    <meta property="og:image" content="{% thumbnail plugins.0.picture 300x170 crop upscale subject_location=plugins.0.picture.subject_location %}">
+    {% get_placeholder_plugins "course_cover" as plugins or %}
+        {{ block.super }}
+    {% endget_placeholder_plugins %}
+    {% blockplugin plugins.0 %}
+        <meta property="og:type" content="website" />
+        {% thumbnail instance.picture 1200x630 crop upscale subject_location=instance.picture.subject_location as thumb %}
+        <meta property="og:image" content="{{ request.scheme }}:{% if not CDN_DOMAIN %}//{{ request.get_host }}{% endif %}{{ thumb.url }}">
```

Comment:
I'm surprised that you need `http:` when `CDN_DOMAIN` is set?

Reply:
Yes, I think the `og:image` requires the full URL. 
On reference documentation https://ogp.me/ and on forums, all the examples start with `http:` or `https:`.

Reply:
On other PR i've added the [fullstatic templatetag](https://github.com/igobranco/richie/commit/34c107ae9bd1d053d360f53d6389fb007a48316c) but the image thumbnail is completely different, and I think this simple if resolves the problem.
There is an alternative one is to put it to the richie context processor.

Reply:
I wasn't sure so I pulled the branch. In fact, when `CDN_DOMAIN` is set, you will also use a custom storage backend that will return the full path for media files as you can see here: https://github.com/openfun/richie-site-factory/blob/master/sites/funmooc/src/backend/base/storage.py#L50

Here is the diff of the tests I made locally:
```diff
diff --git a/src/richie/apps/courses/templates/courses/cms/course_detail.html b/src/richie/apps/courses/templates/courses/cms/course_detail.html
index f7799b01..06b584ff 100644
--- a/src/richie/apps/courses/templates/courses/cms/course_detail.html
+++ b/src/richie/apps/courses/templates/courses/cms/course_detail.html
@@ -17,7 +17,7 @@
     {% blockplugin plugins.0 %}
         <meta property="og:type" content="website" />
         {% thumbnail instance.picture 1200x630 crop upscale subject_location=instance.picture.subject_location as thumb %}
-        <meta property="og:image" content="{{ request.scheme }}:{% if not CDN_DOMAIN %}//{{ request.get_host }}{% endif %}{{ thumb.url }}">
+        <meta property="og:image" content="{% if not CDN_DOMAIN %}{{ SITE.web_url }}{% endif %}{{ thumb.url }}">
         <meta property="og:image:width" content="{{ thumb.width }}">
         <meta property="og:image:height" content="{{ thumb.height }}">
     {% endblockplugin %}
diff --git a/tests/apps/courses/test_templates_course_detail_opengraph.py b/tests/apps/courses/test_templates_course_detail_opengraph.py
index faa31b79..65769492 100644
--- a/tests/apps/courses/test_templates_course_detail_opengraph.py
+++ b/tests/apps/courses/test_templates_course_detail_opengraph.py
@@ -3,6 +3,9 @@ Test suite for the Open Graph for course pages
 """
 import re
 
+from django.test.utils import override_settings
+from django.utils.translation import override
+
 from cms.test_utils.testcases import CMSTestCase
 
 from richie.apps.courses.factories import CourseFactory
@@ -35,6 +38,7 @@ class TemplatesCourseDetailOpengraphCMSTestCase(CMSTestCase):
 
         match = re.search("<meta[^>]+og:image[^>]+>", response_content)
         html_meta_og_image = match.group(0)
+        self.assertIn('content="http://example.com/media', html_meta_og_image)
         self.assertIn(cover_file_name, html_meta_og_image)
 
         match = re.search("<meta[^>]+og:image:width[^>]+>", response_content)
@@ -53,6 +57,34 @@ class TemplatesCourseDetailOpengraphCMSTestCase(CMSTestCase):
             msg="og:image:height should have the maximum permitted value of 630",
         )
 
+    @override_settings(CDN_DOMAIN='xyz.cloudfront.net')
+    def test_open_graph_course_meta_og_image_cdn_domain(self):
+        """
+        Test if the og:image of a course has the correct value image url, width and height on the
+        metas og:image og:image:width and og:image:heigth
+        """
+        cover_file_name = "cover.jpg"
+
+        course = CourseFactory(
+            page_title="Introduction to Programming",
+            code="IntroProg",
+            page_languages=["en", "fr"],
+            fill_cover={
+                "original_filename": cover_file_name,
+            },
+        )
+        course_page = course.extended_object
+        course_page.publish("en")
+
+        url = course_page.get_absolute_url(language="en")
+        response = self.client.get(url)
+        response_content = response.content.decode("UTF-8")
+
+        match = re.search("<meta[^>]+og:image[^>]+>", response_content)
+        html_meta_og_image = match.group(0)
+        self.assertIn('content="/media', html_meta_og_image)
+        self.assertIn(cover_file_name, html_meta_og_image)
+
     def test_open_graph_course_no_cover_image(self):
         """
         Test if a course withthout a cover image has on og:image meta the default logo
@@ -71,4 +103,5 @@ class TemplatesCourseDetailOpengraphCMSTestCase(CMSTestCase):
 
         match = re.search("<meta[^>]+og:image[^>]+>", response_content)
         html_meta_og_image = match.group(0)
+        self.assertIn('content="http://example.com/static', html_meta_og_image)
         self.assertIn("richie/images/logo.png", html_meta_og_image)
(END)
```

Reply:
> In fact, when CDN_DOMAIN is set, you will also use a custom storage backend that will return the full path for media files as you can see here: https://github.com/openfun/richie-site-factory/blob/master/sites/funmooc/src/backend/base/storage.py#L50

It will append the `//<cdn_domain>/media` to the static files, but they want start with `https://<cdn_domain>/media` that the specs requires. I'm going to add a test with the `CDN_DOMAIN` setting similar to the one you sent.

Reply:
I think this code is more obvious:
```html
<meta property="og:image" content="{% if CDN_DOMAIN %}{{ request.scheme }}:{% else %}{{ SITE.web_url }}{% endif %}{{ thumb.url }}">
```

The test is kind of strange, because we need to test with `content="http:/media` because we can't test with the setting `STATICFILES_STORAGE` with the class CDNManifestStaticFilesStorage

Reply:
@sampaccoud why do you have the CDNManifestStaticFilesStorage class ?
https://github.com/openfun/richie-site-factory/blob/master/sites/funmooc/src/backend/base/storage.py#L13-L52

I think it would be better to remove that class from your code and derived if from `STATIC_URL` on the context processor. And use the `STATIC_URL` and `MEDIA_URL` on our Richie apps (Demo, Fun-mooc, NAU, etc.).
Current settings snippet:
```python
    STATIC_URL = "/static/"
    MEDIA_URL = "/media/"
    MEDIA_ROOT = os.path.join(DATA_DIR, "media")
    STATIC_ROOT = os.path.join(DATA_DIR, "static")
```

Proposed settings snippet:
```python
    STATIC_URL = "https://" + os.environ.get("CDN_DOMAIN") + "/static/"
    MEDIA_URL = "https://" + os.environ.get("CDN_DOMAIN") + "/media/"
    MEDIA_ROOT = os.path.join(DATA_DIR, "media")
    STATIC_ROOT = os.path.join(DATA_DIR, "static")
```

Note, the STATIC_URL and MEDIA_URL need to include the protocol.

With this proposed change the code is much more simpler:
```
<meta property="og:image" content="{{ thumb.url }}">
```

What do you think ?